vcf.c

```diff
@@ -451,7 +452,8 @@ int bcf_hdr_register_hrec(bcf_hdr_t *hdr, bcf_hrec_t *hrec)
 
     // INFO/FILTER/FORMAT
     char *id = NULL;
-    int type = -1, num = -1, var = -1, idx = -1;
+    uint32_t type = -1, var = -1;
```

Comment:
Know your C standard!   This conversion is well-defined in section 6.3.1.3:
"Otherwise, if the new type is unsigned, the value is converted by repeatedly adding or
subtracting one more than the maximum value that can be represented in the new type
until the value is in the range of the new type." (with footnote: "The rules describe arithmetic on the mathematical value, not the value of a given type of expression.")

Although it's really down to how this handles invalid input, which is a whole other can o'worms.